src/qgis_plugin_dev_tools/build/rewrite_imports.py

```diff
@@ -115,10 +116,33 @@ def rewrite_imports_in_source_file(
         flags=re.M,
     )
 
-    # add a note describing changes made to files
-    change_note = "# original source code changed by rewriting import statements\n"
-    if change_note not in contents:
-        contents = change_note + contents
+    # special case for custom widgets in .ui files
+    if source_file.suffix == ".ui":
+        ui_tree = ElementTree.fromstring(contents)  # noqa: SC200
+        for widget_section in ui_tree.iter("customwidget"):
+            header_section = widget_section.find("header")
+
+            if header_section is None or header_section.text is None:
+                continue
+
+            if header_section.text == rewritten_package_name:
+                header_section.text = header_section.text.replace(
+                    rewritten_package_name,
+                    f"{container_package_name}.{rewritten_package_name}",
+                )
+            elif header_section.text.startswith(rewritten_package_name + "."):
+                header_section.text = header_section.text.replace(
+                    rewritten_package_name,
```

Comment:
this need some safety not to replace inside other parts of the string, if the same name for reason appears as a part elsewhere?

maybe `header_section.text = f"{container_package_name}.{header_section.text}"` could work?

Reply:
or actually could handle both these at the same time? `if header_section.text == rewritten_package_name or header_section.text.startwith(f"{rewritten_package_name}."): ...`

Reply:
yeah, that's much better, changed as suggested and added one more case to unit test